application/src/main/java/org/thingsboard/server/service/state/DefaultDeviceStateService.java

```diff
@@ -507,32 +514,34 @@ void updateInactivityStateIfExpired(long ts, DeviceId deviceId, DeviceStateData
         }
     }
 
+    private void reportInactivity(long ts, DeviceId deviceId, DeviceStateData stateData) {
+        DeviceState state = stateData.getState();
+        state.setActive(false);
+        state.setLastInactivityAlarmTime(ts);
```

Comment:
We need to make sure that we do not overwrite device state with the stale data. What if the inactivityTs < lastActivityTs?

Reply:
Fixed in https://github.com/thingsboard/thingsboard/pull/9030/commits/e275678c22e4ba89eaca353117c140150fbdb016

Reply:
Seems that the issue is not fixed yet. I don't see a check for the condition: lastInactivityTime is greater than lastActivityTime.

Reply:
It needs to be greater than current lastActivityTime and current lastInactivityTime

Reply:
Good point, fixed in https://github.com/thingsboard/thingsboard/pull/9030/commits/46dca971e119c031f63cd5fab40390d6cf9576a0